src/hooks/useActiveProject.tsx

```diff
@@ -6,164 +6,87 @@ import React, {
-) => {
-  const getDefault = () => {
-    const defaultProject = projects?.[0];
-    const defaultSubject = subjects?.find(
-      (s) => s.projectId === defaultProject?.id,
-    );
-    return { selectedProject: defaultProject, selectedSubject: defaultSubject };
-  };
-
-  if (!projectId) {
-    return getDefault();
-  }
-
-  const selectedProject = projects?.find((p) => p.id === projectId);
-  const selectedSubject = subjects?.find((s) => s.projectId === projectId);
-  if (!selectedProject || !selectedSubject) {
-    if (process.env.NODE_ENV !== 'test') {
-      console.warn('Ignoring attempt to set invalid projectId', projectId);
-    }
-    return getDefault();
-  }
-  return { selectedProject, selectedSubject };
-};
-
 export const ActiveProjectContextProvider = ({
   children,
 }: {
   children?: React.ReactNode;
 }) => {
-  const projectsResult = useSubjectProjects();
-  const projectLoading = projectsResult.isLoading || !projectsResult.isFetched;
+  const { userConfiguration, isLoaded } = useSession();
+  const { user, configurations } = userConfiguration;
+  const { account } = useActiveAccount();
 
-  const useMeResult = useMe();
-  const useMeLoading = useMeResult.isLoading || !useMeResult.isFetched;
-  const { data: userData } = useUser();
-  const userId = userData?.id;
   const [selectedId, setSelectedId] = useState<string>();
-  const [previousUserId, setPreviousUserId] = useState(userId);
+  const configuration = configurations.find((c) => c.account === account?.id);
+
   const [storedProjectIdResult, setStoredProjectId, isStorageLoaded] =
-    useAsyncStorage(
-      `${selectedProjectIdKey}:${userId}`,
-      !!selectedProjectIdKey && !!userId,
-    );
+    useAsyncStorage(`${selectedProjectIdKey}:${user.id}`, isLoaded);
 
   /**
    * Initial setting of activeProject
    */
-  const hookReturnValue = useMemo<ActiveProjectProps>(() => {
-    if (
-      !userId || // wait for user id before reading and writing to storage
-      projectLoading || // wait for projects endpoint
-      useMeLoading ||
-      !isStorageLoaded
-    ) {
-      return {};
-    }
-
-    const { selectedProject, selectedSubject } = findProjectAndSubjectById(
-      selectedId ?? storedProjectIdResult,
-      projectsResult.data,
-      useMeResult.data,
-    );
-
-    if (selectedProject && selectedSubject) {
-      return {
-        activeProject: selectedProject,
-        activeSubjectId: selectedSubject.subjectId,
-        activeSubject: selectedSubject,
-      };
+  const hookReturnValue = useMemo<ActiveSubjectProject | undefined>(() => {
+    if (!isLoaded || !isStorageLoaded) {
+      return undefined;
     }
 
-    return {};
+    const projectToSelect = selectedId ?? storedProjectIdResult;
+    const subjectAndProject = projectToSelect
+      ? configuration?.subjects.find((s) => s.projectId === projectToSelect)
+      : configuration?.subjects[0];
+    return subjectAndProject ? { activeSubject: subjectAndProject } : undefined;
```

Comment:
I feel like it might be better to just return `subjectAndProject`. The API looks a little funky right now `useActiveProject` returning a single value called `activeSubject`